jme3-core/src/plugins/java/com/jme3/texture/plugins/TGALoader.java

```diff
@@ -463,29 +502,139 @@ public static Image load(InputStream in, boolean flip) throws IOException {
 
 
         in.close();
+
+        return resizeImage(width, height, pixelDepth, format, rawData);
+    }
+
+    /**
+     * Change this image if need.
+     *
+     * @param width current width of image
+     * @param height current height of image
+     * @param pixelDepth pixel depth of image
+     * @param format format of image.
+     * @param rawData data of image.
+     * @return result image.
+     */
+    protected Image resizeImage(int width, int height, final int pixelDepth, final Format format, byte[] rawData) {
```

Comment:
Although resizing images automatically could be useful, I don't think this functionality should only be available in certain loaders. This is much better to be implemented as a `TextureProcessor` feature instead.